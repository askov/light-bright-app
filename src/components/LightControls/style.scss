.light-controls {
  margin-left: auto;
}
.control-button {
  cursor: pointer;
  border: none;
  padding: 10px;
  margin-right: 10px;
  transition-duration: 0.5s;
  text-transform: uppercase;
  background-color: #f2f2f2;
  color: #222;
  &:hover {
    background-color: #f44242;
    color: white;
  }
  &:last-child {
    margin-right: 0;
  }
}