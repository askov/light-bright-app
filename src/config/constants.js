export default {
  DEFAULT_LIGHT_COLORS: ['red', 'green', 'yellow', 'blue'],
}