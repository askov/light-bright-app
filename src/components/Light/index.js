import React from 'react';
import PropTypes from 'prop-types';
import './style.scss';

function Light(props) {
  return (
    <div className="light"
         onMouseEnter={() => props.handleLightEnter(props.index)}
         onMouseDown={() => props.handleLightClick(props.index)}
         onDoubleClick={() => props.handleDoubleClick(props.index)}
         style={{backgroundColor: props.color}}>
    </div>
  );
}

Light.propTypes = {
  index: PropTypes.number.isRequired,
  color: PropTypes.string.isRequired,
  handleLightEnter: PropTypes.func.isRequired,
  handleLightClick: PropTypes.func.isRequired,
  handleDoubleClick: PropTypes.func.isRequired,
};

export default Light;
