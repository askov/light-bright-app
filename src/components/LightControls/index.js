import React from 'react';
import PropTypes from 'prop-types';
import './style.scss';

function LightControls(props) {
  return (
    <div className="header">
    <button onClick={props.handleResetAll}>Reset all</button>
    <button onClick={props.handleReset}>Reset</button>
    </div>
  );
}

LightControls.propTypes = {
  handleResetAll: PropTypes.func.isRequired,
  handleReset: PropTypes.func.isRequired,
};

export default LightControls;
