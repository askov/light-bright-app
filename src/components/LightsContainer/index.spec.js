import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import LightContainer from './index';
import Light from '../Light';
import LightControls from '../LightControls';
import config  from '../../config';

describe('<LightContainer />', () => {
  let props;
  let wrapper;

  const createWrapper = (props) => {
    return shallow(<LightContainer {...props} />);
  }

  beforeEach(() => {
    props = {
      handleReset: jest.fn(),
      handleResetAll: jest.fn(),
    };
    wrapper = createWrapper(props);
  });

  test('renders correctly', () => {
    const tree = renderer
      .create(<LightContainer {...props} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders one <LightControls /> component', () => {
    expect(wrapper.find(LightControls)).toHaveLength(1);
  });

  it('renders predefined number of <Light /> components', () => {
    expect(wrapper.find(Light)).toHaveLength(config.constants.LIGHT_QUANTITY);
  });

});