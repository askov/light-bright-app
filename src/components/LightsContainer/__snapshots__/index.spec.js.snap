// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LightContainer /> renders correctly 1`] = `
Array [
  <div>
    Light-Bright App
    <div
      className="light-controls"
    >
      <button
        className="control-button"
        onClick={[Function]}
      >
        Reset all
      </button>
      <button
        className="control-button"
        onClick={[Function]}
      >
        Reset
      </button>
    </div>
  </div>,
  <div
    className="lights-container"
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
  >
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
    <div
      className="light"
      onDoubleClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "backgroundColor": "grey",
        }
      }
    />
  </div>,
]
`;
